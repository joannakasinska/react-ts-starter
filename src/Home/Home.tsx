import React, { PureComponent, ComponentClass } from 'react';
import { connect } from 'react-redux';
import { InjectedIntl, injectIntl } from 'react-intl';

interface HomeStateProps {
    intl: InjectedIntl;
}

export class Home extends PureComponent<HomeStateProps, {}> {
    render() {
        return (
            <section>{this.props.intl.formatMessage({ id: 'hello' })}</section>
        );
    }
}

export const ConnectedHome = connect<HomeStateProps, null>(
    null,
    null
)(injectIntl(Home as ComponentClass<any>)) as ComponentClass<any>;
