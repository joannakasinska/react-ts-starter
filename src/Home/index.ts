import homeRoutes from './routes';

export { homeRoutes };
