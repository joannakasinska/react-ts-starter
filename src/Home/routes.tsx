import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { ConnectedHome } from './Home';

export default () => {
    return (
        <Switch>
            <Route exact path={'/'} component={ConnectedHome} />
        </Switch>
    );
};
