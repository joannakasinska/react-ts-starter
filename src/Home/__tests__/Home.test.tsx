import React from 'react';
import renderer from 'react-test-renderer';
import { createInjectedIntl } from '../../__tests__/mocks';
import { Home } from '../Home';

const props = {
    intl: createInjectedIntl(),
};

describe('Home:', () => {
    it('should render correctly and match snapshot', () => {
        const tree = renderer.create(<Home {...props} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
