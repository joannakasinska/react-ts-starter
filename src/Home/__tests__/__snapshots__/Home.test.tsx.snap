// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home: should render correctly and match snapshot 1`] = `
<section>
  Cześć!
</section>
`;
